src/navigator.js

```diff
@@ -0,0 +1,154 @@
+// navigator.js
+
```

Comment:
Suggest: Add a comment that documents the URI patterns (query strings) that this supports.